Update cheatsheet for Clojure 1.10

jafingerhut opened the pull request:
Here is what I have already added that is new in Clojure 1.10:

Added to Namespace / From symbol section:
    requiring-resolve
Added to Java Interoperation / Exceptions section:
    ex-cause ex-message clojure.main/ex-str

And here are the new things in Clojure 1.10 that have not been added.
If someone would like them added to the cheatsheet, please recommend a
category (or multiple categories) to add them to, whether existing or
new.

    PrintWriter-on
    read+string
    add-tap
    remove-tap
    tap>
    clojure.core.protocols/Datafiable
    clojure.core.protocols/Navigable
    clojure.core.protocols/datafy
    clojure.core.protocols/nav
    clojure.core.server/io-prepl
    clojure.core.server/prepl
    clojure.core.server/remote-prepl
    clojure.core.specs.alpha/even-number-of-forms?
    clojure.spec.gen.alpha/shuffle
    clojure.datafy/datafy
    clojure.datafy/nav
    clojure.main/renumbering-read

- [y] Have you read the [guidelines for contributing](https://clojure.org/community/contributing_site)?
- [y] Have you signed the Clojure Contributor Agreement?
- [y] Have you verified your asciidoc markup is correct?

puredanger commented on content/api/cheatsheet.html:
```diff
@@ -1988,19 +1999,19 @@ <h2>Spec (<a href="https://clojure.org/about/spec">rationale</a>, <a href="https
 
 Initially set to boolean value of clojure.spec.check-asserts
-system property. Defaults to false.">check-asserts</a> <a href="https://clojuredocs.org/clojure_core/clojure.spec/check-asserts_q" title="clojure.spec.alpha/check-asserts?
+system property. Defaults to false.">check-asserts</a> <a href="https://clojuredocs.org/clojure_core/clojure.spec.alpha/check-asserts_q" title="clojure.spec.alpha/check-asserts?
```
link text should be `check-asserts?`

jafingerhut replied:
Good catch.  I'll see if I can figure out why my code for generating this stuff does that, and fix it.

jafingerhut replied:
Actually, I think it may be correct, but please double-check me.  `check-asserts` and `check-asserts?` appear one after the other as entries in the sheet, and they both appear to have the correct link text.

jafingerhut replied:
You can check a different variant of the sheet rendered to HTML here: https://jafingerhut.github.io/cheatsheet/clojuredocs/cheatsheet-tiptip-cdocs-summary.html

It seems to have both check-asserts and check-asserts? with correct links.

puredanger replied:
Oh, I think I just misread the html, nvm

jafingerhut commented:
Definitely open to adding more things here, especially if you have a preference for how it ought to be categorized and can suggest where to add it.

puredanger commented:
the datafy stuff is a new top-level namespace `clojure.datafy` so that's probably a new section that I expect will grow a bit in the future. 

the clojure.main functions could be a new section on error handling tools and include `clojure.core/Throwable->map`, `clojure.main/ex-triage`,`clojure.main/ex-str`, `clojure.main/err->msg`, `clojure.main/report-error` (1.10.1).

Also, the tap stuff should be a new line under IO (`add-tap`, `tap>`, `remove-tap`).

jafingerhut commented:
I added a second commit to this PR with most of the changes you suggested.

You suggested that I add Throwable->map and some related functions into a new section on error handling tools.  I had already added some of them to an existing section "Java Interoperation / Exceptions" earlier, so I added the rest of them to that section with this commit.

If you would also like them in another new section, or in a different section instead of that one, let me know and I can take a look into it.

jafingerhut commented:
Here is a link to the change log explaining exactly what changed in v49, the most recent version, in case you want to compare it to your suggestions.
